tests/src/buildApplication/generateFunctionsMap.test.ts

```diff
@@ -1,105 +1,75 @@
-							? 'middleware.js'
-							: 'index.js',
-					});
-				}
-				if (/validTest\/functions\/.*\/\.vc-config\.json/.test(file)) {
-					return JSON.stringify({
-						runtime:
-							file.includes('should-be-valid') && !file.includes('is-valid')
-								? 'nodejs'
-								: 'edge',
-						entrypoint: 'index.js',
-					});
-				}
-				return '';
-			},
-			mkdir: async () => null,
-			writeFile: async () => null,
-			stat: async (path: string) => {
-				// NOTE: Invalid file is used to (in a hacky way) simulate an function called `middlewarejs` that has an entry point of `middleware.js` being changed to `index.js`, and that new entry point not existing in the file system. Hence, it would be an invalid file, and `isDirectory()` and `isFile()` will then be false.
-				const invalidFile =
-					path.includes('invalidTest') &&
-					path.includes('middlewarejs') &&
-					path.endsWith('.js');
-				const isFile = path.endsWith('.js') || path.endsWith('.json');
-
-				return {
-					isDirectory: () => !invalidFile && !isFile,
-					isFile: () => !invalidFile && isFile,
-				};
-			},
-			readdir: async (rawDir: string) => {
-				const dir = normalizePath(rawDir);
-				if (['validTest/functions', 'invalidTest/functions'].includes(dir)) {
-					return [
-						'api',
-						'index.func',
-						'index.rsc.func',
-						'middlewarejs.func',
-						'base/middleware.func',
-						'path/(group-1)/to/(group-2)/page.func',
-						'(is-valid)/should-be-valid.func', // valid
-						'should-be-valid.func', // invalid
-						'should-be-valid-alt.func', // invalid
-						'rsc/(is-valid)/should-be-valid.func', // valid
-						'rsc/should-be-valid.rsc.func', // invalid
-					];
-				}
-				if (
-					['validTest/functions/api', 'invalidTest/functions/api'].includes(dir)
-				) {
-					return ['hello.func'];
-				}
-				return [];
-			},
-		};
-	});
+const validIndexVcConfigJson = JSON.stringify({
+	runtime: 'edge',
+	entrypoint: 'index.js',
 });
-
-afterAll(() => {
-	vi.clearAllMocks();
+const invalidIndexVcConfigJson = JSON.stringify({
+	runtime: 'nodejs',
+	entrypoint: 'index.js',
 });
 
+const validFuncDir = {
+	'.vc-config.json': validIndexVcConfigJson,
+	'index.js': '',
+};
+
+const invalidFuncDir = {
+	'.vc-config.json': invalidIndexVcConfigJson,
+	'index.js': '',
+};
+
 describe('generateFunctionsMap', async () => {
 	test('should generate a valid functions map (without experimentalMinify), accounting for invalid root-level functions', async () => {
-		const { invalidFunctions, functionsMap } = await generateFunctionsMap(
-			'validTest/functions',
-			false
-		);
+		mockFs({
+			functions: {
+				'index.func': validFuncDir,
+				'index.rsc.func': validFuncDir,
+				api: {
+					'hello.func': validFuncDir,
+				},
+				'middlewarejs.func': {
+					'.vc-config.json': JSON.stringify({
+						name: 'middleware',
+						runtime: 'edge',
+						entrypoint: 'middleware.js',
+					}),
+					'index.js': '', // is this correct? @james-elicx
```

Comment:
This directory should also have a `middleware.js` file present which we ignore, but I think it's worth still listing it to make it a bit clearer that it exists.

I believe the case we're not currently testing now is where it targets an uncompiled `middleware.js`, and there is no compiled `index.js` present. So, adding another function that targets `middleware.js` with only a `middleware.js` file present, that should fail, and then I believe we'll have tested all cases. This can probably be added in the invalid functions tests I guess.